Eatery/Controllers/Eateries/CollegetownEateriesMenuViewController.swift

```diff
@@ -0,0 +1,389 @@
+//
+//  CollegetownMenuViewController.swift
+//  Eatery
+//
+//  Created by Gonzalo Gonzalez on 3/3/19.
+//  Copyright © 2019 CUAppDev. All rights reserved.
+//
+
+import Crashlytics
+import MapKit
+import UIKit
+
+class CollegetownEateriesMenuViewController: UIViewController, UIScrollViewDelegate, MKMapViewDelegate, CLLocationManagerDelegate {
+    
+    var eatery: CollegetownEatery
+    var delegate: MenuButtonsDelegate?
+    var userLocation: CLLocation?
+    
+    var outerScrollView: UIScrollView!
+    var backButton: UIButton!
+    var ctownMenuHeaderView: CollegetownMenuHeaderView!
+    var informativeViews: [UIView]!
+
+    var mapView: MKMapView
+    let eateryAnnotation = MKPointAnnotation()
+    var locationManager: CLLocationManager!
+    
+    var defaultCoordinate: CLLocationCoordinate2D {
+        return eatery.location.coordinate
+    }
+    
+    var midpointCoordinate: CLLocationCoordinate2D {
+        let currentCoordinates = locationManager.location?.coordinate ?? CLLocation.olinLibrary.coordinate
+        let eateryCoordinates = eatery.location.coordinate
+
+        let cLon = currentCoordinates.longitude * .pi / 180
+        let cLat = currentCoordinates.latitude * .pi / 180
+        let eLon = eateryCoordinates.longitude * .pi / 180
+        let eLat = eateryCoordinates.latitude * .pi / 180
+        
+        let dLon = eLon - cLon
+        
+        let x = cos(eLat) * cos(dLon)
+        let y = cos(eLat) * sin(dLon)
+        let centerLatitude = atan2(sin(cLat) + sin(eLat), sqrt((cos(cLat) + x) * (cos(cLat) + x) + y * y))
+        let centerLongitude = cLon + atan2(y, cos(cLat) + x)
+        
+        var midpointCoordinates = CLLocationCoordinate2D()
+        midpointCoordinates.latitude = centerLatitude * 180 / .pi
+        midpointCoordinates.longitude = centerLongitude * 180 / .pi
+        
+        return midpointCoordinates
+    }
+    
+    var deltaLatLon: [Double] {
+        let currentCoordinates = locationManager.location?.coordinate ?? CLLocation.olinLibrary.coordinate
+        let eateryCoordinates = eatery.location.coordinate
+        
+        var deltaLatLon = [Double]()
+        deltaLatLon.append(abs(eateryCoordinates.latitude - currentCoordinates.latitude))
+        deltaLatLon.append(abs(eateryCoordinates.longitude - currentCoordinates.longitude))
+        
+        return deltaLatLon
+    }
+    
+    var maxDeltaLatLon: [Double] {
```

Comment:
Array literal

Reply:
Will leave as variable for new implementation with extensions to function properly